examples/Raytracer_Benchmark/RaytraceBenchmark.cu

```diff
@@ -153,11 +188,56 @@ void initiliazeCudaWorld(cuda::RaytracerData_t *rtdata) {
   rtdata->fVPstate = vpstate;
   rtdata->fWorld   = gpu_world;
 
+  vecgeom::Vector<vecgeom::LogicalVolume *> *container;
+  cudaMallocManaged(&container, sizeof(vecgeom::Vector<vecgeom::LogicalVolume *>));
+
+  getAllLogicalVolumes<<<1,1>>>((vecgeom::VPlacedVolume *)gpu_world, container);
+
+  cudaDeviceSynchronize();
+
+  AttachRegions<<<1,1>>>(volume_container, container);
+}
+
+
+// Print information about containers
+__global__ void print(cuda::RaytracerData_t *rtdata, int index) {
+  auto *vect = &rtdata->sparse_rays[index];
+
+  printf("=== vect: fNshared=%lu/%lu fNused=%lu fNbooked=%lu - shared=%.1f%% sparsity=%.1f%%\n", vect->size(),
+         vect->capacity(), vect->size_used(), vect->size_booked(), 100. * vect->get_shared_fraction(),
+         100. * vect->get_sparsity());
+}
+
+// Print information about containers
+void print_vector_cuda(cuda::RaytracerData_t *rtdata, int no_generations) {
+  for (int i = 0; i < no_generations; ++i)
+    print<<<1,1>>>(rtdata, i); 
 }
 
-int executePipelineGPU(const vecgeom::cxx::VPlacedVolume *world, int argc, char *argv[])
+// Check if there are rays in containers
+__global__ void check_containers(cuda::RaytracerData_t *rtdata, int no_generations, bool *value)
+{
+  for (int i = 0; i < no_generations; ++i) {
```

Comment:
You should probably have: `*value = false;` to start with, not having to care if the user of the function initialized it or not.